view/preview/index.js

```diff
@@ -0,0 +1,148 @@
+  "easeOutQuint",
+  "easeOutSine",
+  "easeOutExpo",
+  "easeOutCirc",
+  "easeOutBack",
+  "easeOutElastic",
+  "easeInQuad",
+  "easeInCubic",
+  "easeInQuart",
+  "easeInQuint",
+  "easeInSine",
+  "easeInExpo",
+  "easeInCirc",
+  "easeInBack",
+  "easeInElastic",
+  "easeInQuad",
+  "easeInOutCubic",
+  "easeInOutQuart",
+  "easeInOutQuint",
+  "easeInOutSine",
+  "easeInOutExpo",
+  "easeInOutCirc",
+  "easeInOutBack",
+  "easeInOutElastic",
+];
+
+const circles = Array.from({ length: CIRCLES }, (_, i) => {
+  const circle = document.createElementNS(SVG_DOCTYPE, "circle");
+  const isMobile = checkIsMobile();
+
+  const strokeWidth = isMobile
+    ? 100 / (i * i) //
+    : 40 / (i * i);
+  const strokeRadius = isMobile
+    ? 70 / Math.log(i / 2) - 15 //
+    : 70 / Math.log(i) - 15;
+
+  circle.setAttribute("stroke-width", `${strokeWidth}vw`);
+  circle.setAttribute("stroke", COLORS[anime.random(0, COLORS.length - 1)]);
+  circle.setAttribute("fill", "none");
+  circle.setAttribute("r", `${strokeRadius}vw`);
+  circle.setAttribute(
+    "stroke-dasharray",
+    `
+      ${anime.random(0, 200)}
+      ${anime.random(0, 200)}
+      ${anime.random(0, 200)}`
+  );
+
+  circle.style.transformOrigin = `${ORIGIN.x} ${ORIGIN.y}`;
+
+  return circle;
+});
+
+// setup fadein
+const fadeIn = anime.timeline({ autoplay: true });
+circles.map((el) => {
+  fadeIn.add(
+    {
+      targets: el,
+      opacity: 1,
+      duration: anime.random(2000, 5000),
+      easing: ANIMATION[anime.random(0, ANIMATION.length - 1)],
+    },
+    0
+  );
+});
+
+// setup animation
+const timeline = anime.timeline({
+  autoplay: true,
+  loop: true,
+  direction: "alternate",
+});
+circles.map((el, i) => {
+  timeline.add(
+    {
+      targets: el,
+      scale: anime.random(MIN_SCALE, MAX_SCALE),
+      rotate: "360deg",
+      duration: anime.random(10000, 30000),
+      easing: ANIMATION[anime.random(0, ANIMATION.length - 1)],
+    },
+    0
+  );
+});
+
+// add circles to html
+const app = document.querySelector(SELECTORS.SCROLLER);
+const svg = document.createElementNS(SVG_DOCTYPE, "svg");
+svg.classList.add(CLASSNAMES.GRAPHCIS);
+app.appendChild(svg);
+circles.map((circle) => svg.appendChild(circle));
+
+// add onScroll logic
+useScroll({
+  element: document.querySelector(SELECTORS.WRAPPER),
+  onStartScroll: timeline.pause,
+  onScroll: (scrollPosition) => {
+    // timeline.seek(timeline.duration * (scrollPosition * 0.01));
+
+    const current = timeline.duration * (timeline.progress * 0.01);
+    const scroll = timeline.duration * (scrollPosition * 0.01);
+    timeline.seek((current + scroll) / 2);
+  },
+  onEndScroll: timeline.play,
+});
+
+// add onResize logic
+function onResize() {
+  const { vw, vh } = getDimensions();
+  setSvgPosition({ svg, width: vw, heigth: vh });
+  circles.map((circle) => {
+    circle.setAttribute("cx", ORIGIN.x);
+    circle.setAttribute("cy", ORIGIN.y);
+  });
+}
+onResize();
+window.addEventListener("resize", onResize);
```

Comment:
💡 
Lets add class/style with [`animation-play-state: pause`](https://developer.mozilla.org/en-US/docs/Web/CSS/animation-play-state) rule if animation not in viewport

Reply:
animejs already did it:) found accidentally in inspector